### DISCUSSION TITLE
GitHub ClassRoom Moodle Error 500

### DISCUSSION THREAD

**Discussion by fabiogongorafreire** [community/community#72219] [GitHub Education]
### Hi everyone,

Hello everyone.
I'm Fabio. I'm working for an university in Spain and we're trying to use github classRoom with moodle.
We followed all steps of the documentation: https://docs.github.com/es/education/manage-coursework-with-github-classroom/teach-with-github-classroom/register-a-learning-management-system-with-github-classroom
We registered the LMS, created and set the external tool in moodle, created the organization and some assignments. But when we select the pre configured tool we named "GitHub ClassRoom" and click on "Select content" we are redirected to https://classroom.github.com/lti1p3/openid-connect/redirect and we get error 500: Internal server error/We're sorry, but something went wrong./Try going back to classroom dashboard](https://classroom.github.com)
We noticed there are other posts related to error 500 but they are not exactly what's happening to us.
We appreciate any help

**Comment by jmerinogranizo** [comment]
Hello everyone.
We have the same problem.
We have done several tests and we have even set our organization as public to avoid permissions problems.
Our Moodle version is 4.1.5.
We would appreciate any idea.
Thanks in advance.

**Comment by jaume77** [comment]
Hello!
We join the list...
We have followed the configuration manuals indicated in the first post, and error 500 also appears...
If anyone has made any progress, the information is welcome.
In our case, the link is with Moodle 4.1.1.
Greetings!

**Comment by EdWright75** [comment]
Hi

I'm also having an issue connecting a course with an "external tool" activity in Moodle.

I've done the "[Register an LMS for GitHub classroom](https://docs.github.com/es/education/manage-coursework-with-github-classroom/teach-with-github-classroom/register-a-learning-management-system-with-github-classroom)" and followed the instructions again (after re-setting the Moodle settings). I then go to the course that I want to be able to link the Classroom as an external tool. I can see my GitHub Classroom external tool but when I click on the ***Select Content*** button, I get an error message as below:

> Error connecting to GitHub Classroom
> Classroom may be misconfigured on your LMS. Contact your LMS administrator.
> Error: Missing required information: Context Membership URL

I've looked through the documentation and searched the forums for more guidance but I can't see anywhere that suggests how to proceed, so, again I ask can anyone help out here please? I'd love to be integrating Moodle and GitHub classroom!

Thanks!

**Reply by najasoft** [reply]
Hi,  same issue as @EdWright75  , moodle version: 4.1.6

**Reply by mgmodell** [reply]
I, too, am experiencing this problem. I'm not an admin and do not have access to do any testing. I'm sure there's a misconfiguration, but what can I ask my admins to look at to determine the solution?

**Reply by NaniChkhenkeli** [reply]
same issue

**Comment by attilastrba** [comment]
I have the same error Moodle Version [Moodle](http://moodle.org/) [4.3.3+ (Build: 20240315)](https://moodledev.io/general/releases)

**Comment by moodleautograding** [comment]
I also have the same issue with Moodle 4.3.3

Error connecting to GitHub Classroom
Classroom may be misconfigured on your LMS. Contact your LMS administrator.

Error: Missing required information: Context Membership URL

**Reply by moodleautograding** [reply]
If we do not check "Supports Deep Linking (Content-Item Message)" it'll resolve the issue.
![image](https://github.com/community/community/assets/156881230/86a8283f-05de-48cb-9312-4d55240b54d7)

**Reply by derUhrheber** [reply]
The [setup instruction](https://github.com/orgs/community/discussions/72219) says this box must be ticked.
How else would you ensure the course - classroom connection?

**Reply by creamcollar1122** [reply]
even if i uncheck it shows same screen for students it says , how can we show assignment in moodle to studnents . Authorization Error in GitHub Classroom
You are not authorized to make this request. Please contact your administrator.

Error: Expected one of the following Roles:

http://purl.imsglobal.org/vocab/lis/v2/system/person#Administrator
http://purl.imsglobal.org/vocab/lis/v2/system/person#AccountAdmin
http://purl.imsglobal.org/vocab/lis/v2/system/person#Creator
http://purl.imsglobal.org/vocab/lis/v2/system/person#SysAdmin
http://purl.imsglobal.org/vocab/lis/v2/system/person#SysSupport
http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator
http://purl.imsglobal.org/vocab/lis/v2/institution/person#Faculty
http://purl.imsglobal.org/vocab/lis/v2/institution/person#Staff
http://purl.imsglobal.org/vocab/lis/v2/institution/person#Instructor
http://purl.imsglobal.org/vocab/lis/v2/membership#Administrator
http://purl.imsglobal.org/vocab/lis/v2/membership#ContentDeveloper
http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor
Your roles:

http://purl.imsglobal.org/vocab/lis/v2/membership#Learner

**Reply by srawat-coursera** [reply]
I am also facing the same when open the link as learner, i have already linked the classroom in my LMS using Administrator account.

**Comment by McDowell-CSSD** [comment]
I'm getting the same issue here on 4.4.2. I am the site admin.

```
Error connecting to GitHub Classroom
Classroom may be misconfigured on your LMS. Contact your LMS administrator.

Error: Missing required information: Context Membership URL
```

**Comment by mastnym** [comment]
We are on 4.3.8 having the same issue. When I initially setup this tool it worked for a while. I tried to tweak some additional settings and it even broke to error 500 sometime. Now I'm getting ` Error: Missing required information: Context Membership URL ` even if I revert all the settings to their defaults. Any help or clues appreciated @github :)

**Comment by sajid264** [comment]
Same problem: Internal Server Error 500 Moodle 4.5.5

**Comment by mrevans-parishhill** [comment]
Same problem:  Error: Missing required information: Context Membership URL

Moodle 5.0.2+

**Comment by Essogbe** [comment]
Seems like Moodle doesn't work anymore with Github Classroom

**Comment by IgnacioPorte** [comment]
hey any news on this, I am having the same problem

**Comment by Erika06** [comment]
Same problem with Moodle 4.5.7
_500 Internal server error. We're sorry, but something went wrong.
[Try going back to classroom dashboard](https://classroom.github.com/)_
